ugvc/mrd/srsnv_plotting_utils.py

```diff
@@ -926,151 +925,108 @@ def plot_qual_per_feature(
         )
 
 
-def get_mixed_data(
+def get_data_subsets(
     df: pd.DataFrame,
+    is_mixed_flag: bool,
 ):
     """generates subsets of the input df by category: mixed/non mixed * cycle-skip/non cycle-skip
 
     Parameters
     ----------
     df : pd.DataFrame
         data set with features, labels and quals of the model
+    is_mixed_flag : bool
+        indicates if the data set contains mixed reads
 
     Returns
     -------
-    df_mixed_cs: pd.DataFrame
-        subset of input df: mixed and cycle-skip
-    df_mixed_non_cs: pd.DataFrame
-        subset of input df: mixed and non cycle-skip
-    df_non_mixed_non_cs: pd.DataFrame
-        subset of input df: non mixed and non cycle-skip
-    df_non_mixed_cs: pd.DataFrame
-        subset of input df: non mixed and cycle-skip
+    df_dict: dict
+        dict of data subsets with features, labels and quals of the model
     """
-    df_mixed_cs = df[(df["is_mixed"] & df["is_cycle_skip"])]
-    df_mixed_non_cs = df[(df["is_mixed"] & ~df["is_cycle_skip"])]
-    df_non_mixed_non_cs = df[(~df["is_mixed"] & ~df["is_cycle_skip"])]
-    df_non_mixed_cs = df[(~df["is_mixed"] & df["is_cycle_skip"])]
-    return df_mixed_cs, df_mixed_non_cs, df_non_mixed_non_cs, df_non_mixed_cs
-
-
-def get_fpr_recalls_mixed(
-    df_mixed_cs: pd.DataFrame,
-    df_mixed_non_cs: pd.DataFrame,
-    df_non_mixed_cs: pd.DataFrame,
-    df_non_mixed_non_cs: pd.DataFrame,
+
+    df_dict = {}
+
+    if is_mixed_flag:
+        df_dict["mixed cycle skip"] = df[(df["is_mixed"] & df["is_cycle_skip"])]
+        df_dict["mixed non cycle skip"] = df[(df["is_mixed"] & ~df["is_cycle_skip"])]
+        df_dict["non mixed non cycle skip"] = df[(~df["is_mixed"] & ~df["is_cycle_skip"])]
+        df_dict["non mixed cycle skip"] = df[(~df["is_mixed"] & df["is_cycle_skip"])]
+    else:
+        df_dict["cycle skip"] = df[df["is_cycle_skip"]]
+        df_dict["non cycle skip"] = df[~df["is_cycle_skip"]]
```

Comment:
In general I think it makes more sense to set these as additional binary columns, not duplicate the data to different dicts, but it's fine to leave as is.